### PR TITLE
chore: Added tests for leave workspace flow

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description

Added and updated tests for the leave workspace flow. 

Fixes # (issue)
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

- Manual
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java [hunk 1]
```diff
@@ -186,10 +203,16 @@ public void leaveWorkspace_WhenUserExistsInWorkspace_RemovesUser() {
...
-                    assertThat(uniqueUsersInWorkspaceBefore).containsAll(Set.of(currentUser.getId(), usertest.getId()));
+                    assertThat(uniqueUsersInWorkspaceBefore).containsAll(Set.of(api_user.getId(), usertest.getId()));
                     assertThat(uniqueUsersInWorkspaceAfter).containsAll(Set.of(usertest.getId()));
                 })
                 .verifyComplete();
+
+        StepVerifier.create(userDataService.getForCurrentUser())
+                .assertNext(userData -> {
+                    assertThat(userData.getRecentlyUsedWorkspaceIds()).doesNotContain(testWorkspace.getId());
+                })
```
Comment (nsarupr) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java:214]: Can you please add an assertion on `recentlyUserApplicationIds`?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java [hunk 2]
```diff
@@ -148,28 +149,44 @@ private void addRolesToWorkspace(List<UserRole> roles) {
     @Test
     @WithUserDetails(value = "api_user")
     public void leaveWorkspace_WhenUserExistsInWorkspace_RemovesUser() {
+        Workspace myWorkspace = new Workspace();
+        myWorkspace.setName("Test org" + UUID.randomUUID());
```
Comment (nsarupr) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java:153]: Can you please use the same naming convention that you have used for the Application?

**Review by nsarupr — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java (hunk 2 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java]: Done

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java]: Added

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java [hunk 3]
```diff
@@ -186,10 +205,17 @@ public void leaveWorkspace_WhenUserExistsInWorkspace_RemovesUser() {
...
-                    assertThat(uniqueUsersInWorkspaceBefore).containsAll(Set.of(currentUser.getId(), usertest.getId()));
+                    assertThat(uniqueUsersInWorkspaceBefore).containsAll(Set.of(api_user.getId(), usertest.getId()));
                     assertThat(uniqueUsersInWorkspaceAfter).containsAll(Set.of(usertest.getId()));
                 })
                 .verifyComplete();
+
+        StepVerifier.create(userDataService.getForCurrentUser())
+                .assertNext(userData -> {
+                    assertThat(userData.getRecentlyUsedWorkspaceIds()).doesNotContain(testWorkspace.getId());
+                    assertThat(userData.getRecentlyUsedAppIds()).doesNotContain(application.getId());
```
Comment (nsarupr) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserWorkspaceServiceTest.java:216]:
```suggestion
                    assertThat(userData.getRecentlyUsedAppIds()).doesNotContain(savedApplication.getId());
```

**Review by nsarupr — APPROVED** [review]
Looks good to me

**Pull request merged**